src/auth/index.ts

```diff
@@ -0,0 +1,394 @@
+      await to(fetch(options.uri, requestOptions));
+    if (fetchError || !fetchResponse) {
+      return {
+        error: fetchError,
+        message: 'Network or CORS error occured',
+      } as FetchError;
+    }
+    if (!fetchResponse.ok) {
+      return {
+        status: fetchResponse.status,
+        message: fetchResponse.statusText,
+        error: new Error(await fetchResponse.text()),
+      } as FetchError;
+    }
+    const [parseError, json] = await to(fetchResponse.json());
+    if (parseError) {
+      return {
+        error: parseError,
+        message: 'Returned data is not valid json',
+      } as FetchError;
+    }
+    const jwt = json as JWTPayload;
+    addApiTokens(jwt);
+    return jwt;
+  };
+
+  return {
+    addListener,
+    eventTrigger,
+    getStatus,
+    getError,
+    getStoredUser,
+    setStoredUser,
+    setStatus,
+    setError,
+    isInitialized,
+    isAuthenticated,
+    fetchApiToken,
+    getApiTokens,
+    addApiTokens,
+    removeApiToken,
+  };
+}
+
+const config: ClientConfig = {
+  realm: getEnv('REACT_APP_OIDC_REALM'),
+  url: getEnv('REACT_APP_OIDC_URL'),
+  authority: getEnv('REACT_APP_OIDC_URL'),
+  clientId: getEnv('REACT_APP_OIDC_CLIENT_ID'),
+  callbackPath: getEnv('REACT_APP_OIDC_CALLBACK_PATH'),
+  logoutPath: getEnv('REACT_APP_OIDC_LOGOUT_PATH'),
+  responseType: getEnv('REACT_APP_OIDC_RESPONSE_TYPE'),
+  scope: getEnv('REACT_APP_OIDC_SCOPE'),
+  tokenExchangePath: getEnv('REACT_APP_OIDC_TOKEN_EXCHANGE_PATH'),
+  autoSignIn: false,
+  hasApiTokenSupport: true,
+  enableLogging: getBooleanEnv('REACT_APP_OIDC_ENABLE_LOGGING'),
+};
+
+export function isCallbackUrl(url: string): boolean {
+  return url === config.callbackPath;
+}
+
+export function getClientConfig(): ClientConfig {
+  return config;
+}
+
+export function hasValidClientConfig(): boolean {
+  return !!(config && config.url && config.clientId);
+}
+
+export function getLocationBasedUri(
+  property: string | undefined
+): string | undefined {
+  const location = window.location.origin;
+  if (property === undefined) {
+    return undefined;
+  }
+  return `${location}${property}`;
+}
+
+export function getTokenUri(clientConfig: ClientConfig): string {
+  if (clientConfig.tokenExchangePath) {
+    return `${clientConfig.url}${clientConfig.tokenExchangePath}`;
+  }
+  return `${clientConfig.url}/realms/${clientConfig.realm}/protocol/openid-connect/token`;
+}
+
+export function createClientGetOrLoadUserFunction({
+  getUser,
+  isInitialized,
+  init,
+}: Pick<Client, 'getUser' | 'isInitialized' | 'init'>): () => Promise<
+  User | undefined | null
+> {
+  return () => {
+    const currentUser = getUser();
+    if (currentUser) {
+      return Promise.resolve(currentUser);
+    }
+    if (isInitialized()) {
+      return Promise.resolve(undefined);
+    }
+    return new Promise((resolve, reject) => {
+      init()
+        .then(() => resolve(getUser()))
+        .catch(e => reject(e));
```

Comment:
Can be simplified to `return init() .then(() => getUser());` from line 388 to 393